controllers/helmreleaseproxy/helmreleaseproxy_controller.go

```diff
@@ -51,11 +53,27 @@ type HelmReleaseProxyReconciler struct {
 func (r *HelmReleaseProxyReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	clusterToHelmReleaseProxies, err := util.ClusterToTypedObjectsMapper(mgr.GetClient(), &addonsv1alpha1.HelmReleaseProxyList{}, mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
 		For(&addonsv1alpha1.HelmReleaseProxy{}).
-		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(log, r.WatchFilterValue)).
+		Watches(
+			&clusterv1.Cluster{},
+			handler.EnqueueRequestsFromMapFunc(clusterToHelmReleaseProxies),
+			builder.WithPredicates(
+				predicates.All(ctrl.LoggerFrom(ctx),
+					predicates.Any(ctrl.LoggerFrom(ctx),
+						predicates.ClusterUnpaused(ctrl.LoggerFrom(ctx)),
+						predicates.ClusterControlPlaneInitialized(ctrl.LoggerFrom(ctx)),
```

Comment:
Hmm no I see that wouldn't work because the `ClusterUnpaused` explicitly requires the cluster to have previously been in unpaused state before to continue on update, which wouldn't be the case just following cluster control plane initialized.

Reply:
From the move logs you can see the order that causes this issue:

```
Creating Cluster="self-hosted-18gxrd" Namespace="self-hosted-wokpvl"
...
Creating HelmChartProxy="cluster-autoscaler-self-hosted-18gxrd" Namespace="self-hosted-wokpvl"
...
Creating HelmReleaseProxy="cluster-autoscaler-self-hosted-18gxrd-7q6p4" Namespace="self-hosted-wokpvl"
```

By the time the `HelmReleaseProxy` has been moved, a new `HelmReleaseProxy` has been created by reconciling the cluster which has had control plane initialized condition set as part of the move.

Reply:
Created #188